server/stream.go

```diff
@@ -2418,6 +2418,12 @@ func (mset *stream) processInboundMirrorMsg(m *inMsg) bool {
 		return false
 	}
 
+	// If the TTL header is set, remove it.
+	hdr := m.hdr
+	if len(hdr) > 0 {
```

Comment:
I think instead of removing we flag that the messages is sourced to this function and if so we ignore this plus a bunch of others..